plugins/inventory/aws_ec2.py

```diff
@@ -176,6 +184,9 @@
     separator: '-'       # Hostname will be aws-test_literal
     prefix: 'aws'
 
+# Returns all the hostnames for a given instance
+allow_duplicated_hosts: False
```

Comment:
```suggestion
allow_duplicated_hosts: True
```

Reply:
My understanding was the the new behaviour is a bug and we would like to switch back to what we've had in 4.0.0.

Reply:
The fix in this PR https://github.com/ansible-collections/amazon.aws/pull/862 should do this (that is how it should be, it's not a bug really).

1. Fix returned hosts when
```
hostnames:
- tag:Tag1,Tag2
```
2. Fix returned hosts when
```
hostnames:
- tag:Tag1
- tag:Tag2
```
3. Fix returned hosts when
```
hostnames:
- tag:Tag1=Test1,Tag2=Test2
```

Given and EC2 instance with the following tags
```
tags:
  Tag1: Test1
  Tag2: Test2
```
Instead of only returning
```
"aws_ec2": {
    "hosts": [
        "Test1"
    ]
},
"tag_Name_instance_02": {
    "hosts": [
        "Test1"
    ]
},
"tag_Tag1_Test1": {
    "hosts": [
        "Test1"
    ]
},
"tag_Tag2_Test2": {
    "hosts": [
        "Test1"
    ]
}
```
It returns now

```
{
"_meta": {
   "hostvars": {
         "Test1": {
                "ami_launch_index": 0,
                "ansible_host": "10.210.0.101",
                   ...},
         "Test2": {
                 "ami_launch_index": 0,
                 "ansible_host": "10.210.0.101",
                   ...},
    },
    "all": {
        "children": [
            "aws_ec2",
            "tag_Name_instance_02",
            "tag_Tag1_Test1",
            "tag_Tag2_Test2",
            "ungrouped"
        ]
    },
    "aws_ec2": {
        "hosts": [
            "Test1",
            "Test2"
        ]
    },
    "tag_Name_instance_02": {
        "hosts": [
            "Test1",
            "Test2"
        ]
    },
    "tag_Tag1_Test1": {
        "hosts": [
            "Test1",
            "Test2"
        ]
    },
    "tag_Tag2_Test2": {
        "hosts": [
            "Test1",
            "Test2"
        ]
    }
}

The problem is it altered the way the hosts were returned introducing a breaking change for the users unfortunately. So, for some users this behaviour is useful while for some others it is not. What about @tremble's suggestion here https://github.com/ansible-collections/amazon.aws/pull/862#issuecomment-1233997139 ?

Reply:
This PR implements what @markuman describe here https://github.com/ansible-collections/amazon.aws/pull/862#issuecomment-1233758859

Reply:
I might just be confused. I would expect `allow_duplicated_hosts: False` to only return unique names.  And I would expect what markuman suggested (`unique_hostnames: true`) to do the same thing.

In this example the comment reads:
`# Returns all the hostnames for a given instance` 
But the example parameter is `allow_duplicated_hosts: False` - which I think would not return "all" hostnames, because it would only return unique / non-duplicate names.

Is my logic wrong?